ts/src/phemex.ts

```diff
@@ -2324,10 +2324,34 @@ export default class phemex extends Exchange {
             // 'posSide': Position direction - "Merged" for oneway mode , "Long" / "Short" for hedge mode
         };
         const clientOrderId = this.safeString2 (params, 'clOrdID', 'clientOrderId');
-        const stopLoss = this.safeValue (params, 'stopLoss');
-        const stopLossDefined = (stopLoss !== undefined);
-        const takeProfit = this.safeValue (params, 'takeProfit');
-        const takeProfitDefined = (takeProfit !== undefined);
+        let stopLoss = this.safeValue (params, 'stopLoss');
+        let stopLossDefined = (stopLoss !== undefined);
+        let takeProfit = this.safeValue (params, 'takeProfit');
+        let takeProfitDefined = (takeProfit !== undefined);
+        // backwards compatibility for sl/tp - migrate them into unified format
+        const stopLossPrice = this.safeNumber (params, 'stopLossPrice');
+        const takeProfitPrice = this.safeString (params, 'takeProfitPrice');
+        if (stopLossPrice !== undefined) {
+            if (stopLossDefined) {
+                throw new InvalidOrder (this.id + ' createOrder() - you have conflicting settings, please use either "stopLoss" unified param or old "stopLossPrice" param');
+            }
+            params = this.omit (params, 'stopLossPrice');
+            stopLoss = {
+                'triggerPrice': stopLossPrice,
+            };
+            stopLossDefined = true;
+        }
+        if (takeProfitPrice !== undefined) {
+            if (takeProfitDefined) {
+                throw new InvalidOrder (this.id + ' createOrder() - you have conflicting settings, please use either "takeProfit" unified param or old "takeProfitPrice" param');
+            }
+            params = this.omit (params, 'takeProfitPrice');
+            takeProfit = {
+                'triggerPrice': takeProfitPrice,
+            };
+            takeProfitDefined = true;
+        }
+        // end of support sl/tp
```

Comment:
this above block is moved from bottom to here, so we can "accept" old format and transform into new stoploss format